### ISSUE TITLE
zone file update: unexpected keyword argument 'dust_included'

### ISSUE THREAD

**Issue by larrysalibra** [blockstack/blockstack-core#685]
Trying to update zone file using the browser wallet payment key:

body of the `PUT` is

```
{"owner_key":"2882b3624b68fe0d3bc10007fcc863a9bc26a725b15801ae76d69fcb20b7298101","zonefile":"$ORIGIN test.id\n$TTL 3600\n_http._tcp URI 10 1 \"https://gaia.blockstack.org/hub/1DjCL3ohUZE8DZvcpuoY6U9EFDz9A55oJU/9/profile.json\"\n","payment_key":"af4f47f2ce6f08b9d0a60ab1e646bc36d6eb96cab652fb40891e4042ec905ad901"}
```

```
[2017-11-20 22:22:18,668] [DEBUG] [registrar:1043] (15386.123145367498752) Registrar sleeping for 1
[2017-11-20 22:22:19,677] [DEBUG] [registrar:1043] (15386.123145367498752) Registrar sleeping for 1
[2017-11-20 22:22:19,886] [DEBUG] [rpc:3927] (15386.123145371705344) 
full path: /v1/names/test.id/zonefile
method: OPTIONS
path: /v1/names/test.id/zonefile
qs: {}
headers:
origin: http://localhost:3000
accept-language: en-US,en;q=0.9
accept-encoding: gzip, deflate, br
access-control-request-headers: authorization,content-type
connection: keep-alive
accept: */*
user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/62.0.3202.94 Safari/537.36
access-control-request-method: PUT
host: localhost:6270
referer: http://localhost:3000/profiles/0/zone-file

[2017-11-20 22:22:19,886] [DEBUG] [rpc:390] (15386.123145371705344) No authorization header
[2017-11-20 22:22:19,887] [DEBUG] [rpc:3969] (15386.123145371705344) No authentication needed
127.0.0.1 - - [20/Nov/2017 22:22:19] "OPTIONS /v1/names/test.id/zonefile HTTP/1.1" 200 -
[2017-11-20 22:22:19,992] [DEBUG] [rpc:3927] (15386.123145371705344) 
full path: /v1/names/test.id/zonefile
method: PUT
path: /v1/names/test.id/zonefile
qs: {}
headers:
origin: http://localhost:3000
content-length: 314
accept-language: en-US,en;q=0.9
accept-encoding: gzip, deflate, br
connection: keep-alive
accept: application/json
user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/62.0.3202.94 Safari/537.36
host: localhost:6270
referer: http://localhost:3000/profiles/0/zone-file
content-type: application/json
authorization: bearer blockstack_integration_test_api_password

[2017-11-20 22:22:19,993] [DEBUG] [app:152] (15386.123145371705344) Not a valid token
[2017-11-20 22:22:19,993] [DEBUG] [rpc:3974] (15386.123145371705344) Authenticated with password
[2017-11-20 22:22:19,998] [DEBUG] [rpc:5511] (15386.123145371705344) RPC running (/tmp/blockstack-run-scenario.blockstack_integration_tests.scenarios.portal_test_env/client/api_endpoint.pid)
[2017-11-20 22:22:20,005] [DEBUG] [actions:262] (15386.123145371705344) Zonefile is not a serialized JSON string; try parsing as text
[2017-11-20 22:22:20,145] [DEBUG] [actions:1713] (15386.123145371705344) Update test.id, zonefile=$ORIGIN test.id
$TTL 3600
_http._tcp URI 10 1 "https://gaia.blockstack.org/hub/1DjCL3ohUZE8DZvcpuoY6U9EFDz9A55oJU/9/profile.json"
, zonefile_hash=caf8ee9ec1643d244952fa9296974e2358e1701d
[2017-11-20 22:22:20,150] [DEBUG] [rpc:5013] (15386.123145371705344) Connect to API at localhost:6270
[2017-11-20 22:22:20,276] [DEBUG] [registrar:1367] (15386.123145371705344) async_update(test.id, zonefile_data=$ORIGIN test.id
$TTL 3600
_http._tcp URI 10 1 "https://gaia.blockstack.org/hub/1DjCL3ohUZE8DZvcpuoY6U9EFDz9A55oJU/9/profile.json"
, profile=None, transfer_address=None)
[2017-11-20 22:22:20,764] [DEBUG] [registrar:1043] (15386.123145367498752) Registrar sleeping for 1
[2017-11-20 22:22:21,042] [DEBUG] [safety:1360] (15386.123145371705344) Check update on test.id: test is_payment_address_usable
[2017-11-20 22:22:21,217] [DEBUG] [session:189] (15386.123145371705344) Connect to bitcoind at localhost:18332 (/tmp/blockstack-run-scenario.blockstack_integration_tests.scenarios.portal_test_env/client/client.ini)
[2017-11-20 22:22:21,218] [DEBUG] [session:104] (15386.123145371705344) [15404] Connect to bitcoind at http://blockstack@localhost:18332, timeout=300.0
[2017-11-20 22:22:21,221] [DEBUG] [fees:93] (15386.123145371705344) Bitcoin estimatefee(2) is 5.5e-05 (5 satoshi/byte)
[2017-11-20 22:22:21,858] [DEBUG] [registrar:1043] (15386.123145367498752) Registrar sleeping for 1
[2017-11-20 22:22:21,903] [DEBUG] [safety:757] (15386.123145371705344) Getting UTXOs for mtF9d6tgHafNzgQEYUmuvPMZ7Dar3z31Zq
[2017-11-20 22:22:21,918] [DEBUG] [safety:769] (15386.123145371705344) Balance of mfop7pPNAi25KTDyaw7mWxNJoRaYqtCiL1 is 4999065295 satoshis
[2017-11-20 22:22:21,918] [DEBUG] [safety:795] (15386.123145371705344) Get total operation fees for running 'update' on test.id owned by mtF9d6tgHafNzgQEYUmuvPMZ7Dar3z31Zq paid by mfop7pPNAi25KTDyaw7mWxNJoRaYqtCiL1
[2017-11-20 22:22:21,918] [DEBUG] [safety:799] (15386.123145371705344) Operation update may consume owner inputs of mtF9d6tgHafNzgQEYUmuvPMZ7Dar3z31Zq
[
    {
        "confirmations": 204, 
        "out_script": "76a9148b9c6fb0a867eabbd4df5f7c9a7506789da2b39988ac", 
        "outpoint": {
            "hash": "6187cf3d723119ef65ac617e247694c600b4080e1e80dffcc3b5cae0ab8ed7e0", 
            "index": 1
        }, 
        "transaction_hash": "6187cf3d723119ef65ac617e247694c600b4080e1e80dffcc3b5cae0ab8ed7e0", 
        "value": 11000
    }
]
[2017-11-20 22:22:21,919] [DEBUG] [safety:800] (15386.123145371705344) Operation update may consume payment inputs of mfop7pPNAi25KTDyaw7mWxNJoRaYqtCiL1
[
    {
        "confirmations": 204, 
        "out_script": "76a91403303ef773529aafa7e1a38714a0ca2c1b14f0d688ac", 
        "outpoint": {
            "hash": "6187cf3d723119ef65ac617e247694c600b4080e1e80dffcc3b5cae0ab8ed7e0", 
            "index": 2
        }, 
        "transaction_hash": "6187cf3d723119ef65ac617e247694c600b4080e1e80dffcc3b5cae0ab8ed7e0", 
        "value": 9998130590
    }
]
[2017-11-20 22:22:21,919] [DEBUG] [safety:1235] (15386.123145371705344) Add task update_tx_fee (<function _estimate_update_tx at 0x1116e3a28>)
[2017-11-20 22:22:21,919] [DEBUG] [safety:577] (15386.123145371705344) Queued tasks for update on test.id: get_balance, is_payment_address_usable, update_tx_fee
[2017-11-20 22:22:21,920] [DEBUG] [utils:337] (15386.123145371705344) Start task 'is_payment_address_usable'
[2017-11-20 22:22:21,920] [DEBUG] [utils:283] (15386.123145375911936) Run task <function <lambda> at 0x112031cf8>
[2017-11-20 22:22:21,920] [DEBUG] [utils:337] (15386.123145371705344) Start task 'update_tx_fee'
[2017-11-20 22:22:21,921] [DEBUG] [utils:337] (15386.123145371705344) Start task 'get_balance'
[2017-11-20 22:22:21,921] [DEBUG] [utils:283] (15386.123145380118528) Run task <functools.partial object at 0x1112a0ec0>
[2017-11-20 22:22:21,922] [DEBUG] [utils:344] (15386.123145371705344) Join task 'is_payment_address_usable'
[2017-11-20 22:22:21,922] [DEBUG] [utils:283] (15386.123145384325120) Run task <function _get_balance at 0x11177e8c0>
[2017-11-20 22:22:21,924] [DEBUG] [utils:285] (15386.123145384325120) Task exit <function _get_balance at 0x11177e8c0>
[2017-11-20 22:22:21,932] [WARNING] [blockchain:287] (15386.123145375911936) Address mfop7pPNAi25KTDyaw7mWxNJoRaYqtCiL1 useable with zero confirmations
[2017-11-20 22:22:21,933] [DEBUG] [utils:285] (15386.123145375911936) Task exit <function <lambda> at 0x112031cf8>
[2017-11-20 22:22:21,933] [DEBUG] [utils:344] (15386.123145371705344) Join task 'get_balance'
[2017-11-20 22:22:21,934] [DEBUG] [utils:344] (15386.123145371705344) Join task 'update_tx_fee'
[2017-11-20 22:22:22,236] [DEBUG] [nameops:175] (15386.123145380118528) Try building a NAME_UPDATE with inputs 0-1 of mfop7pPNAi25KTDyaw7mWxNJoRaYqtCiL1
[2017-11-20 22:22:22,237] [ERROR] [nameops:651] (15386.123145380118528) make_outputs() got an unexpected keyword argument 'dust_included'
Traceback (most recent call last):
  File "/Users/larry/git/blockstack-testing/master/blockstack-venv/lib/python2.7/site-packages/blockstack-0.17.0.1-py2.7.egg/blockstack_client/backend/nameops.py", line 617, in estimate_update_tx_fee
    unsigned_tx, num_utxos = make_cheapest_name_update(name, fake_zonefile_hash, fake_consensus_hash, owner_address, utxo_client, payment_address, payment_utxos )
  File "/Users/larry/git/blockstack-testing/master/blockstack-venv/lib/python2.7/site-packages/blockstack-0.17.0.1-py2.7.egg/blockstack_client/backend/nameops.py", line 293, in make_cheapest_name_update
    subsidize=subsidize, tx_fee=tx_fee, dust_included = dust_included )
  File "/Users/larry/git/blockstack-testing/master/blockstack-venv/lib/python2.7/site-packages/blockstack-0.17.0.1-py2.7.egg/blockstack_client/backend/nameops.py", line 178, in make_cheapest_nameop
    unsigned_tx = tx_builder(*tx_args, **tx_kw)
  File "/Users/larry/git/blockstack-testing/master/blockstack-venv/lib/python2.7/site-packages/blockstack-0.17.0.1-py2.7.egg/blockstack_client/tx.py", line 100, in update_tx
    inputs, outputs = tx_update(*args, **kw)
  File "/Users/larry/git/blockstack-testing/master/blockstack-venv/lib/python2.7/site-packages/blockstack-0.17.0.1-py2.7.egg/blockstack_client/operations/update.py", line 150, in make_transaction
    dust_included = dust_included )
TypeError: make_outputs() got an unexpected keyword argument 'dust_included'
[2017-11-20 22:22:22,241] [DEBUG] [utils:285] (15386.123145380118528) Task exit <functools.partial object at 0x1112a0ec0>
[2017-11-20 22:22:22,242] [DEBUG] [safety:1313] (15386.123145371705344) update_tx_fee +6200 satoshis (tx fee)
[2017-11-20 22:22:22,242] [DEBUG] [safety:1327] (15386.123145371705344) Total cost of update is 6200 satoshis
[2017-11-20 22:22:22,366] [DEBUG] [blockchain:46] (15386.123145371705344) Connect to bitcoind at localhost:18332 (/tmp/blockstack-run-scenario.blockstack_integration_tests.scenarios.portal_test_env/client/client.ini)
[2017-11-20 22:22:22,367] [DEBUG] [session:104] (15386.123145371705344) [15404] Connect to bitcoind at http://blockstack@localhost:18332, timeout=300.0
[2017-11-20 22:22:22,394] [DEBUG] [nameops:1570] (15386.123145371705344) Payment address mfop7pPNAi25KTDyaw7mWxNJoRaYqtCiL1 has 1 UTXOs
[2017-11-20 22:22:22,394] [DEBUG] [nameops:1571] (15386.123145371705344) Owner address mtF9d6tgHafNzgQEYUmuvPMZ7Dar3z31Zq has 1 UTXOs
[2017-11-20 22:22:22,394] [DEBUG] [nameops:175] (15386.123145371705344) Try building a NAME_UPDATE with inputs 0-1 of mfop7pPNAi25KTDyaw7mWxNJoRaYqtCiL1
[2017-11-20 22:22:22,395] [ERROR] [nameops:2536] (15386.123145371705344) make_outputs() got an unexpected keyword argument 'dust_included'
Traceback (most recent call last):
  File "/Users/larry/git/blockstack-testing/master/blockstack-venv/lib/python2.7/site-packages/blockstack-0.17.0.1-py2.7.egg/blockstack_client/backend/nameops.py", line 2534, in async_update
    config_path=config_path, proxy=proxy, force_update=force_update )
  File "/Users/larry/git/blockstack-testing/master/blockstack-venv/lib/python2.7/site-packages/blockstack-0.17.0.1-py2.7.egg/blockstack_client/backend/nameops.py", line 1590, in do_update
    unsigned_tx, num_utxos = make_cheapest_name_update(fqu, zonefile_hash, consensus_hash, owner_address, utxo_client, payment_address, payment_utxos, subsidize=True )
  File "/Users/larry/git/blockstack-testing/master/blockstack-venv/lib/python2.7/site-packages/blockstack-0.17.0.1-py2.7.egg/blockstack_client/backend/nameops.py", line 293, in make_cheapest_name_update
    subsidize=subsidize, tx_fee=tx_fee, dust_included = dust_included )
  File "/Users/larry/git/blockstack-testing/master/blockstack-venv/lib/python2.7/site-packages/blockstack-0.17.0.1-py2.7.egg/blockstack_client/backend/nameops.py", line 178, in make_cheapest_nameop
    unsigned_tx = tx_builder(*tx_args, **tx_kw)
  File "/Users/larry/git/blockstack-testing/master/blockstack-venv/lib/python2.7/site-packages/blockstack-0.17.0.1-py2.7.egg/blockstack_client/tx.py", line 100, in update_tx
    inputs, outputs = tx_update(*args, **kw)
  File "/Users/larry/git/blockstack-testing/master/blockstack-venv/lib/python2.7/site-packages/blockstack-0.17.0.1-py2.7.egg/blockstack_client/operations/update.py", line 150, in make_transaction
    dust_included = dust_included )
TypeError: make_outputs() got an unexpected keyword argument 'dust_included'
[2017-11-20 22:22:22,395] [ERROR] [registrar:1390] (15386.123145371705344) async_update failed with: 'Failed to sign and broadcast update transaction. Exception: make_outputs() got an unexpected keyword argument 'dust_included''
[2017-11-20 22:22:22,395] [DEBUG] [actions:1737] (15386.123145371705344) RPC error: Failed to sign and broadcast update transaction. Exception: make_outputs() got an unexpected keyword argument 'dust_included'
[2017-11-20 22:22:22,396] [ERROR] [rpc:1096] (15386.123145371705344) Failed to update test.id: Failed to sign and broadcast update transaction. Exception: make_outputs() got an unexpected keyword argument 'dust_included'
127.0.0.1 - - [20/Nov/2017 22:22:22] "PUT /v1/names/test.id/zonefile HTTP/1.1" 503 -
[2017-11-20 22:22:22,876] [DEBUG] [registrar:1043] (15386.123145367498752) Registrar sleeping for 1
[2017-11-20 22:22:23,888] [DEBUG] [registrar:1043] (15386.123145367498752) Registrar sleeping for 1
[2017-11-20 22:22:24,903] [DEBUG] [registrar:1043] (15386.123145367498752) Registrar sleeping for 1
[2017-11-20 22:22:25,919] [DEBUG] [registrar:1043] (15386.123145367498752) Registrar sleeping for 1
```

**Comment by kantai** [comment]
This was fixed a in v0.17.3. Try with the latest version of the blockstack_client.

**Comment by larrysalibra** [comment]
Thanks @kantai - upgrading fixed it!
